src/main/java/org/mitre/synthea/engine/ExpressedSymptom.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    public Long getLastUpdateTime() {
+      return lastUpdateTime;
+    }
+
+    public void setLastUpdateTime(Long lastUpdateTime) {
+      this.lastUpdateTime = lastUpdateTime;
+    }
+
+    public String getSource() {
+      return source;
+    }
+  }
+
+  //keep track of the different sources of the expressed conditions
+  public Map<String, SymptomSource> sources;
+  private String name;
+  
+  public ExpressedSymptom(String name) {
+    this.name = name;  
+    sources = new ConcurrentHashMap<String, SymptomSource>();
+  }
+  
+  public ExpressedSymptom clone() {
+    ExpressedSymptom data = new ExpressedSymptom(this.name);
+    data.sources.putAll(new ConcurrentHashMap<String, SymptomSource>(this.sources));
+    return data;
+  }
+  
+  /** this method updates the data structure wit a symptom being onset from a module.
+   */
+  public void onSet(String module, String cause, long time, int value, Boolean addressed) {    
+    if (!sources.containsKey(module)) {
+      sources.put(module, new SymptomSource(module));
+    }
+    SymptomInfo info = new SymptomInfo(cause, value);
+    sources.get(module).timeInfos.put(Long.valueOf(time), info);
+    sources.get(module).setLastUpdateTime(Long.valueOf(time));
+    sources.get(module).setStatus(addressed);
+  }
+  
+  /**
+   * Method for retrieving the value associated to a given symptom. 
+   * This correspond to the maximum value across all potential causes.
+   */
+  public int getSymptom() {
+    int max = 0;
+    for (String module : sources.keySet()) {
+      Long lastUpdateTime = sources.get(module).getLastUpdateTime();
+      Boolean status = sources.get(module).isStatus();
+      if (sources.get(module).timeInfos.get(lastUpdateTime).getValue() > max && !status) {
+        max = sources.get(module).timeInfos.get(lastUpdateTime).getValue();
+      }
+    }
+    return max;
+  }
+  
+  /**
+   * Method for retrieving the source with the high value not yet addressed. 
+   */ 
+  public String getSourceWithHighValue() {
+    String result = null;
+    int max = 0;
+    for (String module : sources.keySet()) {
+      Long lastUpdateTime = sources.get(module).getLastUpdateTime();
+      Boolean status = sources.get(module).isStatus();
+      if (result == null && !status) {
+        result = module;
+        max = sources.get(module).timeInfos.get(lastUpdateTime).getValue();
+      } else if (sources.get(module).timeInfos.get(lastUpdateTime).getValue() > max && !status) {
+        result = module;
+        max = sources.get(module).timeInfos.get(lastUpdateTime).getValue();
+      }
+    }
+    return result;
+  }
+  
+  /**
+   * Method for retrieving the value associated to a given source. 
+   */  
+  public Integer getValueFromSource(String source) {
+    if (!sources.containsKey(source)) {
+      return null;
+    }
+    Long lastUpdateTime = sources.get(source).getLastUpdateTime();
+    return sources.get(source).timeInfos.get(lastUpdateTime).getValue();    
```

Comment:
See earlier comment re whether lastUpdateTime should be computed from timeInfos vs stored and updated separately. This could return null - is that OK?